herbert_core/classes/instrument_classes/instrument/instruments/@IX_null_inst/IX_null_inst.m

```diff
@@ -0,0 +1,270 @@
+classdef IX_null_inst < IX_inst
+    % An instrument constructed from a struct with no fields
+    % As the new Experiment class has an array of IX_inst, we need
+    % something derived from IX_inst to hold the case where the file from
+    % which the sqw is created has these empty struct instruments.
+    % This class will provide a conversion to an empty struct.
+    
+    properties
+        % none in addition to the base IX_inst
+    end
+    
+    methods
+        function obj = IX_null_inst()
+            % constructs a vanilla instance based on IX_inst
+            obj = obj@IX_inst();
+        end
+        
+        function str = null_struct(~)
+            %makes the null struct for storage in a file
+            str = struct();
+        end
+    end
+
+    %======================================================================
+    % Methods for fast construction of structure with independent properties
+    methods (Static, Access = private)
+        function names = propNamesIndep_
+            % Determine the independent property names and cache the result.
```

Comment:
```suggestion
            % Determine the independent property names and cache the result.
            persistent names_store
            if isempty(names_store)
                names_store = fieldnamesIndep(eval(mfilename('class')));
            end
            names = names_store;            
```
this is private function specific to the class. eval('somethin') is usually least effective way of doing things and very difficult to debug. 
Should be: -- much simpler and efficient
```
 names_store = fieldnamesIndep(IX_null_inst());
```